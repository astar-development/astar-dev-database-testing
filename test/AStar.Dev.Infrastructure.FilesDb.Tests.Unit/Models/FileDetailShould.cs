using AStar.Dev.Infrastructure.FilesDb.Models;
using AStar.Dev.Utilities;
using Shouldly;

namespace AStar.Dev.Infrastructure.FilesDb.Tests.Unit.Models;

public class FileDetailShould
{
    [Fact]
    public void ContainTheExpectedProperties()
        => new FileDetail
           {
               Id            = new (68),
               DirectoryName = new ("mock-directory-name"),
               FileName      = new ("mock-file-name"),
               ImageDetail   = new (7, 8),
               FileHandle    = new ("mock-file-handle"),
               DeletionStatus =
                   new()
                   {
                       HardDeletePending = new DateTimeOffset(2025, 07, 28, 1, 2, 3, TimeSpan.Zero),
                       SoftDeletePending = new DateTimeOffset(2025, 07, 28, 2, 3, 4, TimeSpan.Zero),
                       SoftDeleted       = new DateTimeOffset(2025, 07, 28, 3, 4, 5, TimeSpan.Zero)
                   },
               FileClassifications = new List<FileClassification>
                                     {
                                         new ()
                                         {
                                             Celebrity = true,
                                             Id        = 1,
                                             Name      = "Celebrity",
                                             FileDetails   = new List<FileDetail>
                                                             {
                                                                 new ()
                                                                 {
                                                                     DirectoryName = new ("mock-directory-name2") ,
                                                                     FileName      = new("mock-file-name"),
                                                                     UpdatedOn     = new (2025, 07, 28, 1, 2, 3, TimeSpan.Zero)
                                                                 }
                                                             },
                                             FileNameParts = new List<FileNamePart> { new () { Text        = "mock-file-name-part" } }
                                         }
                                     },
               FileSize         = 1245,
               IsImage          = true,
               FileCreated      = new (2025, 07, 28, 5,  6,  7,  TimeSpan.Zero),
               FileLastModified = new (2025, 07, 28, 10, 20, 30, TimeSpan.Zero),
               FileLastViewed   = new DateTimeOffset(2025, 07, 28, 11, 22, 33, TimeSpan.Zero),
               UpdatedBy        = "mock-user-name",
               UpdatedOn        = new (2025, 07, 28, 8, 7, 9, TimeSpan.Zero),
               MoveRequired     = true
           }
           .ToJson()
           .ShouldMatchApproved();
}
