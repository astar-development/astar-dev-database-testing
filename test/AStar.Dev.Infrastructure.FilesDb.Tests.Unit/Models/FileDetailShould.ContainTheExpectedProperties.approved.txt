{
  "fileClassifications": [
    {
      "id": 1,
      "name": "Celebrity",
      "fileDetails": [
        {
          "fileClassifications": [],
          "id": {
            "value": 0
          },
          "imageDetail": {
            "width": null,
            "height": null
          },
          "fileName": {
            "value": "mock-file-name"
          },
          "directoryName": {
            "value": "mock-directory-name2"
          },
          "fullNameWithPath": "mock-directory-name2/mock-file-name",
          "fileSize": 0,
          "isImage": false,
          "fileHandle": {
            "value": null
          },
          "fileCreated": "0001-01-01T00:00:00+00:00",
          "fileLastModified": "0001-01-01T00:00:00+00:00",
          "fileLastViewed": null,
          "moveRequired": false,
          "deletionStatus": {
            "softDeleted": null,
            "softDeletePending": null,
            "hardDeletePending": null
          },
          "updatedBy": "Jay Barden",
          "updatedOn": "2025-07-28T01:02:03+00:00"
        }
      ],
      "celebrity": true,
      "fileNameParts": [
        {
          "id": 0,
          "text": "mock-file-name-part",
          "fileClassifications": []
        }
      ]
    }
  ],
  "id": {
    "value": 68
  },
  "imageDetail": {
    "width": 7,
    "height": 8
  },
  "fileName": {
    "value": "mock-file-name"
  },
  "directoryName": {
    "value": "mock-directory-name"
  },
  "fullNameWithPath": "mock-directory-name/mock-file-name",
  "fileSize": 1245,
  "isImage": true,
  "fileHandle": {
    "value": "mock-file-handle"
  },
  "fileCreated": "2025-07-28T05:06:07+00:00",
  "fileLastModified": "2025-07-28T10:20:30+00:00",
  "fileLastViewed": "2025-07-28T11:22:33+00:00",
  "moveRequired": true,
  "deletionStatus": {
    "softDeleted": "2025-07-28T03:04:05+00:00",
    "softDeletePending": "2025-07-28T02:03:04+00:00",
    "hardDeletePending": "2025-07-28T01:02:03+00:00"
  },
  "updatedBy": "mock-user-name",
  "updatedOn": "2025-07-28T08:07:09+00:00"
}